src/caf.js

```diff
@@ -28,7 +28,22 @@ module.exports.signalAll = signalAll;
 
 
 function CAF(generatorFn) {
-	return function instance(tokenOrSignal,...args){
+	// allow users to introspect to understand if the function they're dealing
+	// with was generated by CAF
+	Object.defineProperty(
+		instance,
+		"isCAF",
+		{ value: true },
+	);
+	Object.defineProperty(
+		instance,
+		"generatorName",
+		{ value: generatorFn.name || "anonymous" },
+	);
+
+	return instance;
+
+	function instance(tokenOrSignal,...args){
```

Comment:
if we don't change the `name` property of the function instance, it won't be obvious in console debugging that such a function is CAF-wrapped. it might be easy to miss that fact in your debugging if you don't think to look for the `isCAF` boolean.